engine/lib/mySQLRepository.js

```diff
@@ -0,0 +1,108 @@
+'use strict';
+
+const knex = require('knex')
+const logger = require('./logger')
+const fileUtil = require('./fileUtil')
+
+class DatabaseRepository {
+  constructor(databaseConfiguration, invLoader = () => fileUtil.readJsonFile('data/inventory.json')) {
+    this._connection = undefined
+    this._invLoader = invLoader
+
+    const { host, port, user, password, name } = databaseConfiguration
+    this._configuration = {
+      host,
+      port,
+      user,
+      password,
+      database: name,
+      charset: 'utf8'
+    }
+  }
+
+  async isConnected() {
+    if (!this._connection) {
+      await this.initialize()
+    }
+
+    try {
+      await this._connection.select('1')
+      return true
+    }
+    catch (err) {
+      return false
+    }
+  }
+
+  async initialize() {
+    logger.info('Initializing MySQL database')
+    const tableName = 'inventory'
+    await this.createDatabase()
+
+    const connection = knex({
+      client: 'mysql',
+      connection: this._configuration
+    })
+
+    await connection.schema.dropTableIfExists(tableName)
+
+    logger.info('Creating table with name: inventory')
+    await connection.schema.createTable('inventory', table => {
+      table.string('id')
+      table.string('item')
+      table.string('price')
+      table.string('sku')
+      table.string('description')
+    })
+
+    const inventoryData = this._invLoader()
+    for (const item of inventoryData) {
+      logger.info(`Inserting inventory with id: ${item.id}`)
+      await connection(tableName).insert(item)
+    }
+
+    this._connection = connection
+ }
+
+  async createDatabase() {
+    const config = this._configuration
+
+    const rawConnection = knex({
+      client: 'mysql',
+      connection: {
+        host: config.host,
+        port: config.port,
+        user: config.user,
+        password: config.password 
+      }
+    })
+
+    logger.info(`Creating database with name: ${config.database}`)
+    await rawConnection.raw(`CREATE DATABASE IF NOT EXISTS ${config.database}`)
+    rawConnection.destroy()
+  }
+
+  async findAll() {
+    if (!this._connection) {
+      await this.initialize()
+    }
+
+    const results = await this._connection.select().from('inventory')
+    return results
```

Comment:
This could probably be simplified to just:

```suggestion
    return this._connection.select().from('inventory')
```

Reply:
Done!